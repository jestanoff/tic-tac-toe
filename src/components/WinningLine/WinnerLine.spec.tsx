import * as React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import WinnerLine from './index';

const emptyFn = () => null;

describe('<WinnerLine />', () => {
  test('should render an animated line with color', () => {
    const { asFragment } = render(<WinnerLine color="#fff000" line={3} handleClick={emptyFn} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test('should call handleClick on clicking on the line', async () => {
    const handleClickMock = jest.fn();
    render(<WinnerLine color="#fff000" line={3} handleClick={handleClickMock} />);

    await userEvent.click(screen.getByTestId('winner-line'));
    await waitFor(() => expect(handleClickMock).toHaveBeenCalled());
    expect(handleClickMock).toHaveBeenCalledTimes(1);
    expect(handleClickMock).toHaveBeenCalledWith(expect.any(Object));
  });
});
