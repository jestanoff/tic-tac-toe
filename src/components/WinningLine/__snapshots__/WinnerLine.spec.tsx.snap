// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WinnerLine /> should render an animated line with color 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div>
      <svg
        data-testid="winner-line"
        preserveAspectRatio="xMinYMin meet"
        version="1.1"
        viewBox="0 0 500 500"
      >
        <line
          stroke="#fff000"
          stroke-width="16"
          x1="82"
          x2="82"
          y1="30"
          y2="30"
        >
          <animate
            attributeName="x2"
            attributeType="XML"
            begin="250ms"
            dur="125ms"
            fill="freeze"
            from="82"
            repeatCount="1"
            to="82"
          />
          <animate
            attributeName="y2"
            attributeType="XML"
            begin="250ms"
            dur="125ms"
            fill="freeze"
            from="30"
            repeatCount="1"
            to="470"
          />
        </line>
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
